ext/pcntl/pcntl.c

```diff
@@ -986,7 +991,7 @@ PHP_FUNCTION(pcntl_signal)
 			php_error_docref(NULL, E_WARNING, "Error assigning signal");
 			RETURN_FALSE;
 		}
-		zend_hash_index_del(&PCNTL_G(php_signal_table), signo);
+		zend_hash_index_update(&PCNTL_G(php_signal_table), signo, handle);
```

Comment:
Yes, that was my concern, it is tricky, After thinking a bit more, probably it should be ensured that pcntl_signal_dispatch() doesn't handle an integer as a function. Possible erroneous scenario
```
pcntl_signal(SIGINT, "some_handler"); /* recorded into the lookup table */
/* do something else */
/* get SIGINT from outside, it has landed in the queue, but not yet handled */
pcntl_signal(SIGINT, SIG_IGN); /* the handler is an int */
```
By this state, it could be possible that to the time of dispatching signals, call_user_function gets an integer instead fo callable. Not sure how to phpt test this, would be yet more tricky :) 

Thanks.